src/ssl.c

```diff
@@ -11060,10 +11060,12 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
         ssl->biowr = wr;
 
         /* set SSL to use BIO callbacks instead */
-        if (rd != NULL && rd->type != WOLFSSL_BIO_SOCKET) {
+        if (((ssl->cbioFlag & WOLFSSL_CBIO_RECV) != 0) &&
+            (rd != NULL && rd->type != WOLFSSL_BIO_SOCKET)) {
             ssl->CBIORecv = BioReceive;
         }
-        if (wr != NULL && wr->type != WOLFSSL_BIO_SOCKET) {
+        if (((ssl->cbioFlag & WOLFSSL_CBIO_SEND) != 0) &&
```

Comment:
are these conditions checking the expected case? If the result of cbioFlag & WOLFSSL_CBIO_SEND is not 0 then the user has set custom callbacks. If the user has set custom callbacks then we would not want to override them.